__tests__/app.test.js

```diff
@@ -80,3 +80,25 @@ describe("GET /api/articles/:article_id", () => {
   })
 })
 
+describe("Get /api/articles", ()=>{
+  test("GET 200: responds with an array of articles sorted by descending order", () => {
+    return request(app)
+    .get("/api/articles")
+    .expect(200)
+    .then((response) => {
+      const {articles} = response.body
+      expect(articles).toBeSortedBy("created_at",{ descending: true })
+      articles.forEach((article) => { 
+      expect(article).toHaveProperty("author")
+      expect(article).toHaveProperty("title")
+      expect(article).toHaveProperty("article_id")
+      expect(article).toHaveProperty("topic")
+      expect(article).toHaveProperty("created_at")
+      expect(article).toHaveProperty("votes")
+      expect(article).toHaveProperty("article_img_url")
+      expect(article).toHaveProperty("comment_count")
+      })
+    })
+  })
+})
+
```

Comment:
A few things to note about this test
- You've got all the expect statements inside a forEach. This works, unless you have an empty array for the response body! There's nothing to iterate through so your tests wouldn't get a chance to run (and fail). This means you should check the array length before the forEach to protect about that.

- I would also say just checking the only the keys you expect doesn't really prove that all of the articles have the values you expect also. We also aren't checking what the values are (or what data type they should be).


Consider testing the data type too, with something along the lines of...
```
articles.forEach((article) => {
expect(article).hasOwnProperty("article_id", expect.any(Number))
```
Would make far stronger tests, because you're also making sure the value is the right data type.